cachito/workers/pkg_managers/pip.py

```diff
@@ -1978,12 +1992,11 @@ def _push_downloaded_requirement(requirement, pip_repo_name, raw_repo_name):
         dest_dir, filename = requirement["raw_component_name"].rsplit("/", 1)
         upload_raw_package(raw_repo_name, requirement["path"], dest_dir, filename, True)
         if requirement["kind"] == "vcs":
-            version = (
-                f"vcs:{requirement['host']}/{requirement['namespace']}/{requirement['repo']}@"
-                f"{requirement['ref']}"
-            )
+            # Version is "vcs:" followed by the value for the vcs_url purl qualifier
+            version = f"vcs:git+{requirement['url']}@{requirement['ref']}"
         else:
-            version = requirement["original_url"]
+            # Version is "url:" followed by the value for the download_url purl qualifier
+            version = f"url:{requirement['url_with_hash']}"
```

Comment:
Am I right in thinking that it does not matter what we save in the database as `version` as long as we can properly construct purls from it?

Reply:
Yes!
Note that those versions will be available to the users. However, a user should only care for the ICM data when checking for component versions.

Reply:
Gotcha, thanks!

I might drop the prefixes anyway, there should be no ambiguity because a VCS dep is going to start with `git+<protocol>://`, a URL dep is going to start with `<protocol>://` and PyPI deps will have no scheme

Reply:
Sounds good to me

Reply:
If VCS dep is going to start with `git+<protocol>://` as mentioned here , why have we mentioned the other vsc formats for `VCS_SCHEMES` in https://github.com/release-engineering/cachito/blob/0408601daca7af2e6a4e5f9a4f52ee60f5aa37df/cachito/workers/pkg_managers/pip.py#L921

Reply:
@nirzari  Pip does support all those other schemes, see here: https://pip.pypa.io/en/stable/reference/pip_install/#vcs-support

I'd say it is reasonable to support all those for *parsing* requirements files. However, for *downloading* the requirements, we only support `git`. Any of the other VCS prefixes would cause the build to fail.